utils/terminfo.mli

```diff
@@ -18,9 +18,10 @@
 type status =
   | Uninitialised
   | Bad_term
-  | Good_term of int  (* number of lines of the terminal *)
-;;
-external setup : out_channel -> status = "caml_terminfo_setup";;
-external backup : int -> unit = "caml_terminfo_backup";;
-external standout : bool -> unit = "caml_terminfo_standout";;
-external resume : int -> unit = "caml_terminfo_resume";;
+  | Good_term
+
+val setup : unit -> status
```

Comment:
The old mechanism of caching the channel in C was a bit ugly, but it feels similarly wrong to be hardcoding this to stdout. Given such a "big" change, couldn't this either become a functor or have type `status` converted to a "terminal" or something which gets passed around to the other functions just so that the possibility exists to reuse this for stderr in future?

Reply:
Of course the functions from the `Terminfo` module could (should?) take an `out_channel` so that they are not specific to stdout.  However, stdout is hardcoded one level up, in `Location.highlight_terminfo`, which takes a `ppf: Format.formatter` as parameter, but has no way to extract the underlying `out_channel` (if any).  Hence it assumes that `ppf = Format.stdout` and works from there.